api/src/repositories/history-publish-repository.ts

```diff
@@ -609,10 +613,221 @@ export class HistoryPublishRepository extends BaseRepository {
+
+    return response;
+  }
+
+  /**
+   * Gets the count of unpublished survey reports
+   *
+   * @param {number} surveyId
+   * @return {*}  {Promise<QueryResult>}
+   * @memberof HistoryPublishRepository
+   */
+  async getCountSurveyUnpublishedReports(surveyId: number): Promise<QueryResult> {
+    const sqlStatement = SQL`
+    SELECT
+      count(*)
+    from
+      survey_report_attachment sra
+    left join
+      survey_report_publish srp
+    on
+      sra.survey_report_attachment_id = srp.survey_report_attachment_id
+    where
+      sra.survey_id =${surveyId}
+    and
+      srp.survey_report_publish_id is null;
+  `;
+
+    const response = await this.connection.sql(sqlStatement);
+
+    return response;
+  }
+
+  /**
+   * Gets the latest survey observation that wasn't deleted
+   *
+   * @param {number} surveyId
+   * @return {*}  {Promise<QueryResult>}
+   * @memberof HistoryPublishRepository
+   */
+  async getLatestUndeletedObservationRecordId(surveyId: number): Promise<QueryResult> {
+    const sqlStatement = SQL`
+    select
+      occurrence_submission_id
+    from
+      occurrence_submission os
+    where
+      os.survey_id = ${surveyId}
+    and
+      os.delete_timestamp is null
+    order by
+      event_timestamp desc
+    limit 1;`;
+    const response = await this.connection.sql(sqlStatement);
+
+    console.log('response: ', response);
+
+    return response;
+  }
+
+  /**
+   * Gets a publication record for an observation occurrence submission
+   *
+   * @param {number} occurrenceSubmissionId
+   * @return {*}  {Promise<QueryResult>}
+   * @memberof HistoryPublishRepository
+   */
+  async getConfirmationLatestObservationPublished(occurrenceSubmissionId: number): Promise<QueryResult> {
+    const sqlStatement = SQL`
+    select
+      *
+    from
+      occurrence_submission_publish osp
+    where
+      osp.occurrence_submission_id = ${occurrenceSubmissionId};
+`;
+
+    const response = await this.connection.sql(sqlStatement);
+
+    return response;
+  }
+
+  /**
+   * Gets the latest summary result that wasn't deleted
+   *
+   * @param {number} surveyId
+   * @return {*}  {Promise<QueryResult>}
+   * @memberof HistoryPublishRepository
+   */
+  async getLatestUndeletedSummaryResultsId(surveyId: number): Promise<QueryResult> {
+    const sqlStatement = SQL`
+    select
+      survey_summary_submission_id
+    from
+      survey_summary_submission sss
+    where
+      sss.survey_id = ${surveyId}
+    and
+      sss.delete_timestamp is null
+    order by
+      event_timestamp desc
+    limit 1;
+    `;
+    const response = await this.connection.sql(sqlStatement);
+
+    return response;
+  }
+
+  /**
+   * Gets a publication record for a summary result submission
+   *
+   * @param {number} summarySubmissionId
+   * @return {*}  {Promise<QueryResult>}
+   * @memberof HistoryPublishRepository
+   */
+  async getConfirmationLatestSummaryResultsPublished(summarySubmissionId: number): Promise<QueryResult> {
```

Comment:
I think this is a duplicate of `getSurveySummarySubmissionPublishRecord`

Reply:
swapped out